lib/core/src/msParam.cpp

```diff
@@ -1451,60 +1451,61 @@ int
         return SYS_INTERNAL_NULL_INPUT_ERR;
     }
 
-    if ( msKeyValStr->type == NULL ) {
-        fillStrInMsParam( msKeyValStr, NULL );
+    if (msKeyValStr->type == nullptr) {
+        fillStrInMsParam(msKeyValStr, nullptr);
     }
 
-    char *keyPtr = parseMspForStr( keyStr );
-    int keyLen;
-    if ( keyPtr == NULL || strcmp( keyPtr, MS_NULL_STR ) == 0 ) {
-        keyLen = 0;
-    }
-    else {
-        keyLen = strlen( keyPtr );
-    }
+    auto get_length{[](const char* str) -> size_t {
+        if (str == nullptr || std::strcmp(str, MS_NULL_STR) == 0) {
+	    return 0;
+	}
+	return std::strlen(str);
+    }};
 
-    char *valPtr = parseMspForStr( valStr );
-    int valLen;
-    if ( valPtr == NULL || strcmp( valPtr, MS_NULL_STR ) == 0 ) {
-        valLen = 0;
-    }
-    else {
-        valLen = strlen( valPtr );
-    }
-    if ( valLen + keyLen <= 0 ) {
+    const char *keyPtr{parseMspForStr(keyStr)};
+    int keyLen{static_cast<int>(get_length(keyPtr))};
+
+    const char *valPtr{parseMspForStr(valStr)};
+    int valLen{static_cast<int>(get_length(valPtr))};
+
+    if (valLen + keyLen <= 0) {
         return 0;
     }
 
-    char *oldKeyValPtr = parseMspForStr( msKeyValStr );
+    char *oldKeyValPtr{parseMspForStr(msKeyValStr)};
     char *newKeyValPtr, *tmpPtr;
-    if ( oldKeyValPtr == NULL ) {
-        int newLen = valLen + keyLen + 10;
-        newKeyValPtr = ( char * )malloc( newLen );
+    if (oldKeyValPtr == nullptr) {
+        const int newLen{valLen + keyLen + 10};
+        newKeyValPtr = reinterpret_cast<char*>(std::malloc(newLen));
         *newKeyValPtr = '\0';
         tmpPtr = newKeyValPtr;
     }
     else {
-        int oldLen = strlen( oldKeyValPtr );
-        int newLen = oldLen + valLen + keyLen + 10;
-        newKeyValPtr = ( char * )malloc( newLen );
-        snprintf( newKeyValPtr, newLen, "%s%s", oldKeyValPtr, MS_INP_SEP_STR );
-        tmpPtr = newKeyValPtr + oldLen + 4;
-        free( oldKeyValPtr );
-    }
-
-    if ( keyLen > 0 ) {
-        snprintf( tmpPtr, keyLen + 2, "%s=", keyPtr );
+        const int oldLen{static_cast<int>(std::strlen(oldKeyValPtr))};
+        const int newLen{oldLen + valLen + keyLen + 10};
+        newKeyValPtr = reinterpret_cast<char*>(std::malloc(newLen));
+	if (oldLen > 0) {
+	    std::snprintf(newKeyValPtr, newLen, "%s%s", oldKeyValPtr, MS_INP_SEP_STR);
+	    tmpPtr = newKeyValPtr + oldLen + 4;
+	} else {
+  	    *newKeyValPtr = '\0';
+	    tmpPtr = newKeyValPtr;
+	}
+	std::free(oldKeyValPtr);
+    }
+
+    if (keyLen > 0) {
+        std::snprintf(tmpPtr, keyLen + 2, "%s=", keyPtr);
         tmpPtr += keyLen + 1;
     }
-    if ( valLen > 0 ) {
-        snprintf( tmpPtr, valLen + 2, "%s", valPtr );
+    if (valLen > 0) {
+        std::snprintf(tmpPtr, valLen + 2, "%s", valPtr);
     }
-    msKeyValStr->inOutStruct = ( void * ) newKeyValPtr;
+    msKeyValStr->inOutStruct = reinterpret_cast<void *>(newKeyValPtr);
```

Comment:
You can remove the cast here.

C++ allows pointers to be assigned to void pointers implicitly.

Reply:
Done! Didn't know about that implicit cast.